src/utilities/selection/SelectionZone.tsx

```diff
@@ -217,7 +248,11 @@ export class SelectionZone extends React.Component<ISelectionZoneProps, any> {
       } else if (ev.which === KeyCodes.a && (ev.ctrlKey || ev.metaKey) && selectionMode === SelectionMode.multiple) {
         selection.setAllSelected(true);
       } else if (ev.which === KeyCodes.escape) {
-        selection.setAllSelected(false);
+        if (selection.getSelectedCount() > 0) {
```

Comment:
When we do no work, we are not supposed to cancel the keypress. When we hit escape, without this check, we were stopping propagation even when doing no work. In our products, we need the event to bubble up higher so that we can take a different action (like navigate up a folder when you hit escape with nothing selected.)